### PR TITLE
306 catalogue pages need be updated to use entercoursemodule

### PR DESCRIPTION
**Pull request by SirL3X0N**
A few notes to sooth potential headaches. I initially chose to pass the whole course object instead of the cid as we need three values when deciding what button to show.
commit: https://github.com/eclectic-bytech/webcourses-ngx-v4/commit/11b66e48250f58c44282df683c88a358261aec9d

I decided just to change the text inside of the button instead of using ngif when deciding what button to show.
commit: https://github.com/eclectic-bytech/webcourses-ngx-v4/commit/dca83124380adb1a33deaebb764bf54d626e15d3

When changing Enter to Review I decided to change Resume to Continue. I did this because the options would have been resume review and start. so you can see how the first two may have started to blend together.
 
Finally I had to pass what kind of button we wanted on each page. I don't know if you thought of this initially but we need two versions of each button type. this is because they are meant to look different on the catalogue page as opposed to the about page

### CODE REVIEW

**Review by bytech-ca — CHANGES_REQUESTED** [review]
Missed changes:

- [ ] remove btnSize variable. use `[ngClass]="'wide'"` instead.

New items:

- [ ] rename `buttonName` to `buttonLabel`
- [ ] make buttons in course directories match that from course page (paint them all purple ;) )

**Comment by bytech-ca** [comment]
There are two 306 branches, and it looks like both contain unique commits. From what I can tell, it looks like `306-merge-to-master` might need be merged into `306-catalogue-pages-need-be-updated-to-use-entercoursemodule` (this one has the pull request.)

**Comment by SirL3X0N** [comment]
A brief recap of our conversations regarding this issue. I had originally created this pull request when you redid the organization of the angular pages directory, because of this you told me to make a new branch #306-merge-to-master. I then merged the changes of this issue to that branch. while doing so I discovered a bug we both missed. I patched it on both branches instead of cherry picking (my bad, sorry). Both #306 branches are the same but merge to master uses the new file organization method. I will male a new pull request #315

tldr: you told me to make two branches, merge to master needs to be merged to master #315
